pkg/snapshot/snapshot.go

```diff
@@ -141,10 +142,16 @@ func (s *Snapshotter) snapShotFS(f io.Writer) (bool, error) {
 		// Only add the whiteout if the directory for the file still exists.
 		dir := filepath.Dir(path)
 		if _, ok := memFs[dir]; ok {
-			logrus.Infof("Adding whiteout for %s", path)
-			filesAdded = true
-			if err := util.Whiteout(path, w); err != nil {
-				return false, err
+			maybeAdd, err := s.l.MaybeAddWhiteout(path)
```

Comment:
nit: this boolean should probably be called something like "addWhiteout"

Reply:
done